src/page-modules/assistant/details/index.tsx

```diff
@@ -28,7 +28,7 @@ export function AssistantDetails({ tripPattern }: AssistantDetailsProps) {
   );
 
   const tripSearchParams = router.query.id
-    ? tripQueryStringToQueryParams(String(router.query.id))
+    ? tripQueryStringToQueryParams(router.query)
     : undefined;
```

Comment:
You might disagree with me, but I don't think we should modify `tripQueryStringToQueryParams` without changing the `decompressedQueryString`. What about doing something like this? 

```tsx
  const tripSearchParams = router.query.id
    ? tripQueryStringToQueryParams(String(router.query.id))
    : undefined;

  if (tripSearchParams && router.query.filter) {
    tripSearchParams.append('filter', router.query.filter as string);
  }

```
This only requires us to add the filter as a query param to the href in `/trip-pattern/index.tsx` (already done) and to use it when redirecting back to the travel suggestions (the suggested code above)